models/CMakeLists.txt

```diff
@@ -1,19 +1,30 @@
-# List the subdirectory (http://stackoverflow.com/questions/7787823/cmake-how-to-get-the-name-of-all-subdirectories-of-a-directory)
-macro(SUBDIRLIST result curdir)
-  file(GLOB children RELATIVE ${curdir} ${curdir}/*)
-  set(dirlist "")
-  foreach(child ${children})
-    if(IS_DIRECTORY ${curdir}/${child})
-      list(APPEND dirlist ${child})
-    endif()
-  endforeach()
-  set(${result} ${dirlist})
-endmacro()
-
-# Get list of models 
-subdirlist(subdirs ${CMAKE_CURRENT_SOURCE_DIR})
-
-# Install each model 
-foreach (dir ${subdirs})
-  install (DIRECTORY ${dir} DESTINATION share/gazebo/models)
-endforeach ()
\ No newline at end of file
+# configure_files for all files in a folder
+# https://stackoverflow.com/questions/697560/how-to-copy-directory-from-source-tree-to-binary-tree
+macro(CONFIGURE_FILES srcDir destDir)
+  make_directory(${destDir})
+  file(GLOB_RECURSE templateFiles RELATIVE ${srcDir} ${srcDir}/*)
+
+  foreach(templateFile ${templateFiles})
+    set(srcTemplatePath ${srcDir}/${templateFile})
+    message(${srcTemplatePath})
+
+    if(NOT IS_DIRECTORY ${srcTemplatePath})
+      configure_file(${srcTemplatePath}
+                     ${destDir}/${templateFile}
+                     @ONLY)
+    endif()
```

Comment:
Fixed in https://github.com/robotology-playground/icub-models/commit/6efbd84b0a747bf2a50d0c81d72d89aa6bbc2e89